### PR TITLE
Offer enum completion inside an enum member's initializer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/5419

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
tagging @MattGertz 

**Customer scenario**

Customer is typign inside an enum, but does not get enum autocomplete in an enum initializer.

**Bugs this fixes:**

#5419

**Workarounds, if any**

None

**Risk**

Low

**Performance impact**

Low

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Compiler behaves differently inside an enum, leading our enum completion code to not behave as expected.

**How was the bug found?**

Customer reported.

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProviderTests.cs [hunk 1]
```diff
@@ -518,5 +516,105 @@ enum E
...
+
+        [WorkItem(5419, "https://github.com/dotnet/roslyn/issues/5419")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestInEnumInitializer5()
+        {
+            var markup =
+@"using System;
+
+[Flags]
+internal enum ProjectTreeWriterOptions
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProviderTests.cs]: ❕ Need tests for each of the allowed underlying enum types (prime candidate for a `[Theory]`)

❕ Need a test for an enum without `[Flags]` so future changes in behavior won't go undetected.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Was this already working in VB? (I can imagine it was given how magically different that is.)

**Pull request merged**